application/common/components/S3.php

```diff
@@ -104,40 +106,51 @@ public function readS3File($build, $fileName){
     }
 
     /**
-     * copyS3Folder copies the files from where they have been saved encrypted in s3 by codebuild
+     * copyS3BuildFolder copies the files from where they have been saved encrypted in s3 by codebuild
      * to the final unencrypted artifacts folder. 
      * NOTE: This move is required because the initial codebuild version encrypts the files
      * with a key and there is no option to build them without encryption.
      * 
-     * @param Build $build - The build associated with the successful build
+     * @param Build or Release $uses_artifacts - The build or release
      */
-    public function copyS3Folder($build){
+    public function copyS3Folder($uses_artifacts){
         $artifactsBucket = self::getArtifactsBucket();
-        $sourcePrefix = self::getBasePrefixUrl($build, 'codebuild-output') . "/";
-        $destPrefix = self::getBasePrefixUrl($build, self::getAppEnv()) . "/";
-        $baseS3Key = self::getS3KeyBase($build);
+        $sourcePrefix = $uses_artifacts->getBasePrefixUrl('codebuild-output') . "/";
+        $destPrefix = $uses_artifacts->getBasePrefixUrl(self::getAppEnv()) . "/";
         $publicBaseUrl = $this->s3Client->getObjectUrl($artifactsBucket, $destPrefix);
-        $build->beginArtifacts($publicBaseUrl);
+        $uses_artifacts->beginArtifacts($publicBaseUrl);
+        $destFolderPrefix = $uses_artifacts->getBasePrefixUrl(self::getAppEnv());
+        try
+        {
+            $this->s3Client->deleteMatchingObjects($artifactsBucket, $destFolderPrefix);
```

Comment:
Added comment.  Probably more in testing than otherwise, but hit the condition where the destination folder already existed from a previous run.  In my case, job 1 build 1 had already existed.  In the event of an error, the error file was overwritten, but all the other existing files stayed on through the next run.